bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/internal/ScriptEngineManagerImpl.java

```diff
@@ -76,6 +87,19 @@ public void removeScriptEngineFactory(ScriptEngineFactory provider) {
         for (String language : provider.getLanguages()) {
             this.supportedLanguages.remove(language, provider);
         }
+
+        TreeMap<String, String> languageNames = provider.getLanguageNames();
+
+        for (Map.Entry<String, String> entry : languageNames.entrySet()) {
+            if (this.supportedLanguageNames.containsKey(entry.getKey())) {
+                this.supportedLanguageNames.remove(entry.getKey(), entry.getValue());
+            }
+        }
+    }
+
+    @Override
+    public TreeMap<String, String> getSupportedLanguageNames() {
+        return this.supportedLanguageNames;
```

Comment:
Do not return your internal map if you do not use a non modifiable collection.
The caller can modify the map and this could break you and other consumers.

e.g. `getSupportedLanguageNames().put("foo", "bar");`

Reply:
Done